pkg/pex/pex.go

```diff
@@ -184,18 +183,19 @@ func (px *PeerExchange) Advertise(ctx context.Context, ns string, _ ...discovery
 	}
 
 	// If cache is empty or all peers fail, fall back on boot service.
+	// Try to gossip with peers until TTL times out.
 	peers, err := px.disc.Bootstrap(ctx, px.log.WithField("ns", ns), ns)
 	if err != nil {
 		return 0, err
 	}
 
-	for info := range peers {
-		if err = px.gossipRound(ctx, g, info); err == nil {
-			return ttl, nil
+	go func() {
+		for info := range peers {
+			if err = px.gossipRound(ctx, g, info); err == nil {
+				px.log.Error(err.Error())
```

Comment:
Yes, I will revert it now, and log it in DEBUG level. However, if no peers are found, it is neither an error, since it may be the first node to join the network.